ethereum-consensus/src/deneb/polynomial_commitments.rs

```diff
@@ -0,0 +1,147 @@
+
+#[derive(Debug, Error)]
+pub struct Error(c_kzg::Error);
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // Implement custom formatting logic here
+        write!(f, "Custom error: {:?}", self.0)
+    }
+}
+
+impl From<c_kzg::Error> for Error {
+    fn from(inner_error: c_kzg::Error) -> Self {
+        Error(inner_error)
+    }
+}
+
+pub struct ProofAndEvaluation {
+    pub proof: KzgProof,
+    pub evaluation: Bytes32,
+}
+
+pub fn blob_to_kzg_commitment<const BYTES_PER_BLOB: usize>(
+    blob: &Blob<BYTES_PER_BLOB>,
+    kzg_settings: &KzgSettings,
+) -> Result<KzgCommitment, Error> {
+    let blob = c_kzg::Blob::from_bytes(blob.as_ref())?;
+
+    let commitment = c_kzg::KzgCommitment::blob_to_kzg_commitment(&blob, kzg_settings)?;
+    let inner = ByteVector::try_from(commitment.to_bytes().as_slice()).expect("Correct size");
+    Ok(inner)
+}
+
+pub fn compute_kzg_proof<const BYTES_PER_BLOB: usize>(
+    blob: &Blob<BYTES_PER_BLOB>,
+    evaluation_point: &FieldElement,
+    kzg_settings: &KzgSettings,
+) -> Result<ProofAndEvaluation, Error> {
+    let blob = c_kzg::Blob::from_bytes(blob.as_ref())?;
+    let evaluation_point = c_kzg::Bytes32::from_bytes(evaluation_point.as_ref())?;
+
+    let (proof, evaluation) =
+        c_kzg::KzgProof::compute_kzg_proof(&blob, &evaluation_point, kzg_settings)?;
+    let proof = ByteVector::try_from(proof.to_bytes().as_ref()).expect("Correct size");
+    let evaluation = ByteVector::try_from(evaluation.as_slice()).expect("Correct size");
+
+    let result = ProofAndEvaluation { proof, evaluation };
+    Ok(result)
+}
+
+pub fn compute_blob_kzg_proof<const BYTES_PER_BLOB: usize>(
+    blob: &Blob<BYTES_PER_BLOB>,
+    commitment: &KzgCommitment,
+    kzg_settings: &KzgSettings,
+) -> Result<KzgProof, Error> {
+    let blob = c_kzg::Blob::from_bytes(blob.as_ref())?;
+    let commitment = c_kzg::Bytes48::from_bytes(commitment.as_ref()).expect("Correct size");
+
+    let proof = c_kzg::KzgProof::compute_blob_kzg_proof(&blob, &commitment, kzg_settings)?;
+
+    Ok(KzgProof::try_from(proof.to_bytes().as_ref()).expect("input is correct size"))
+}
+
+pub fn verify_kzg_proof(
+    commitment: &KzgCommitment,
+    evaluation_point: &FieldElement,
+    result_point: &FieldElement,
+    proof: &KzgProof,
+    kzg_settings: &KzgSettings,
+) -> Result<(), Error> {
+    let evaluation_point = c_kzg::Bytes32::from_bytes(evaluation_point.as_ref())?;
+    let result_point = c_kzg::Bytes32::from_bytes(result_point.as_ref())?;
+    let commitment = c_kzg::Bytes48::from_bytes(commitment.as_ref()).expect("Correct size");
+    let proof = c_kzg::Bytes48::from_bytes(proof.as_ref()).expect("Correct size");
+
+    let res = c_kzg::KzgProof::verify_kzg_proof(
+        &commitment,
+        &evaluation_point,
+        &result_point,
+        &proof,
+        kzg_settings,
+    )?;
+
+    res.then_some(()).ok_or(Error(c_kzg::Error::InvalidKzgProof(String::from("Invalid Proof"))))
```

Comment:
if you go see how these are used in the code, these variants `InvalidKzg{Proof, Commitment}` are just used to signal that the size of the input data is the wrong

so two things here:
1) you are overloading the meaning of this term in our code, which is likely to confuse other readers (including yourself in the future)
2) we also limit the scope of how much the `c_kzg` types leak out of this module by not reusing them to forward on to others

to fix these:

let's extend the `Error` type you have above to an enum with variants for all of our possible errors

```rust
enum Error {
    #[error(transparent)]
    CKzg(c_kzg::Error),
    #[error("proof verification failed")]
    InvalidProof,
}
```

Reply:
`Error::InvalidProof` suffices for all 3 functions?